### PR TITLE
Bumped version to 2.6

### LINKED ISSUE #421: Release version 2.6.0
This is a component issue for __2.6.0__.
Coming from [opensearch-build__3081__](https://github.com/opensearch-project/opensearch-build/issues/3081). Please follow the following checklist.
__Please refer to the DATES in that post__.

<details><summary>How to use this issue</summary>
<p>

## This Component Release Issue

This issue captures the state of the OpenSearch release, on component/plugin level; its assignee is responsible for driving the release. Please contact them or @mention them on this issue for help. 
Any release related work can be linked to this issue or added as comments to create visiblity into the release status.

## Release Steps

There are several steps to the release process; these steps are completed as the whole component release and components that are behind present risk to the release. The component owner resolves the tasks in this issue and communicate with the overall release owner to make sure each component are moving along as expected.

Steps have completion dates for coordinating efforts between the components of a release; components can start as soon as they are ready far in advance of a future release. The most current set of dates is on the overall release issue linked at the top of this issue.

## The Overall Release Issue

Linked at the top of this issue, the overall release issue captures the state of the entire OpenSearch release including references to this issue, the release owner which is the assignee is responsible for communicating the release status broadly.  Please contact them or @mention them on that issue for help.

## What should I do if my plugin isn't making any changes?

If including changes in this release, increment the version on `__2.x__` branch to `__2.6.0__` for Min/Core, and `__2.6.0.0__` for components. Otherwise, keep the version number unchanged for both.

</p>
</details>

### Preparation

- [x] Assign this issue to a release owner.
- [x] Finalize scope and feature set and update [the Public Roadmap](https://github.com/orgs/opensearch-project/projects/1).
- [x] All the tasks in this issue have been reviewed by the release owner.
- [x] Create, update, triage and label all features and issues targeted for this release with `v2.6.0`.

### CI/CD

- [x] All code changes for `__2.6.0__` are complete.
- [x] Ensure working and passing CI.
- [x] Check that this repo is included in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.6.0).

### Pre-Release

- [x] Update to the `__2.6__` release branch in the [distribution manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/2.6.0).
- [x] Increment the version on the parent branch to the next development iteration.
- [x] Gather, review and publish release notes following the [rules](https://github.com/opensearch-project/opensearch-plugins/blob/main/RELEASE_NOTES.md) and back port it to the release branch.[git-release-notes](https://github.com/ariatemplates/git-release-notes) may be used to generate release notes from your commit history.
- [x] Confirm that all changes for `__2.6.0__` have been merged.
- [x] Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) for the next developer iteration.

### Release Testing

- [x] Find/fix bugs using latest tarball and docker image provided in parent release issue and update the release notes if necessary.
- [x] Code Complete: Test within the distribution, ensuring integration, backwards compatibility, and performance tests pass.
- [x] Sanity Testing: Sanity testing *and* fixing of critical issues found.
- [x] File issues for all intermittent test failures.

### Release

- [x] Complete [documentation](https://github.com/opensearch-project/documentation-website).
- [x] Verify all issued labeled for this release are closed or labelled for the next release.
- [x] Verify the release date mentioned in release notes is correct and matches actual release date.

### Post Release

- [x] Prepare for an eventual security fix development iteration by incrementing the version on the release branch to the next eventual patch version.
- [ ] Add this repo to the [manifest](https://github.com/opensearch-project/opensearch-build/blob/main/manifests/) of the next eventual security patch version.
- [x] [Suggest improvements](https://github.com/opensearch-project/opensearch-build/issues/new) to [this template](https://github.com/opensearch-project/opensearch-build/blob/main/meta/templates/releases/release_template.md).
- [ ] Conduct a retrospective, and publish its results.

### PR DESCRIPTION
**Pull request by amsiglan**
### Description
Bumped version from 2.5 to 2.6.

### Issues Resolved
#421

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by AWSHurneyt — APPROVED** [review]

**Comment by amsiglan** [comment]
@AMoo-Miki The #446 didn't seem to have fixed the yarn mismatch issue. I guess we need 2.6/2.x in OSD to have updated yarn as well?

File: package.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
     "ts-loader": "^6.2.1"
   },
   "engines": {
-    "yarn": "^1.21.1"
+    "yarn": "1.21.1"
```
Comment (AMoo-Miki) [inline package.json:69]: This change should not happen. Please leave it as `^1.21.1`. This plugin would only and only change this version if this plugin needs something from Yarn that the previous version did not offer, and also it has to always be a range and never a specific version.

**Review by AMoo-Miki — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (amsiglan) [reply package.json]: aah, I should have reverted this

**Review by sbcd90 — APPROVED** [review]

File: .github/workflows/cypress-workflow.yml [hunk 2]
```diff
@@ -7,9 +7,9 @@ on:
     branches:
       - "*"
 env:
-  OPENSEARCH_DASHBOARDS_VERSION: '2.5.0'
-  OPENSEARCH_VERSION: '2.5.0-SNAPSHOT'
-  SECURITY_ANALYTICS_BRANCH: '2.5'
+  OPENSEARCH_DASHBOARDS_VERSION: '2.6.0'
```
Comment (AWSHurneyt) [inline .github/workflows/cypress-workflow.yml:10]: You'll want to change this from `2.6.0` to `2.6`. 
`2.6.0` will cause the action to look for a `tag` with that name to use as its `ref`, whereas `2.6` will cause it to look for a `branch` with that name. The `2.6.0` tag isn't available yet as we haven't completed the release.

File: .github/workflows/unit-tests-workflow.yml [hunk 3]
```diff
@@ -7,7 +7,7 @@ on:
     branches:
       - "*"
 env:
-  OPENSEARCH_DASHBOARDS_VERSION: '2.5.0'
+  OPENSEARCH_DASHBOARDS_VERSION: '2.6.0'
```
Comment (AWSHurneyt) [inline .github/workflows/unit-tests-workflow.yml:10]: Same as above; you'll want to change this from `2.6.0` to `2.6`.

File: .github/workflows/cypress-workflow.yml (hunk 2 above)
Comment (amsiglan) [reply .github/workflows/cypress-workflow.yml]: aah, thanks!

File: .github/workflows/unit-tests-workflow.yml (hunk 3 above)
Comment (amsiglan) [reply .github/workflows/unit-tests-workflow.yml]: Why did we use tags then in the first place?

Comment (AWSHurneyt) [reply .github/workflows/unit-tests-workflow.yml]: The tags are helpful when OSD bumps the version on their branches. Often times, they're release branches will be set to an incremental version which doesn't align with what we have on that same release branch.

**Comment by amsiglan** [comment]
The cypress tests are failing due to backend security analytics build failure.

**Comment by amsiglan** [comment]
The tests passed locally, hence merging it.

**Pull request merged**